File: src/server_service.cpp
#include "ros/ros.h"
#include "service_example/Archivo_var.h"

bool filtro_funcion(service_example::Archivo_var::Request &req, service_example::Archivo_var::Response &res)
{
  res.resultado = req.valores[1] * 0.9391 + req.valores[0] * 0.0609;
  ROS_INFO("Resultado del filtrado: [%f]", res.resultado);
  return true;
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "service_server");
  ros::NodeHandle n;
  ros::ServiceServer service = n.advertiseService("filtro", filtro_funcion);
  ROS_INFO("Filtro servidor esta listo!");
  ros::spin();

  return 0;
}

File: src/server_service.py
#!/usr/bin/env python3
import rospy
from service_example.srv import Archivo_var, Archivo_varResponse

def funcion(request): # funcion tipo callback

  filtro = request.valores[1]*0.0609 + request.valores[0]*0.9391
  rospy.loginfo("Resultado del filtrado: "+str(filtro))
  return Archivo_varResponse(filtro) 

if __name__ == "__main__":

  rospy.init_node('service_server')
  server = rospy.Service('filtro', Archivo_var, funcion)
  rospy.loginfo("Filtro servidor esta listo!")
  rospy.spin()

# Archivo_var.srv
# float32[] valores
# ---
# float32 resultado
